tests/parsers/test_audio.py

```diff
@@ -28,3 +28,16 @@ def test_audio(parser, make_tag):
     parsed = parser.parse(tag)[0]
     assert parsed.get('streams')[0]["url"] == 'audiosource'
     assert parsed.get('type') == 'audio'
+
+
+def test_audio_with_spaces_in_src(parser, make_tag):
```

Comment:
It may be a good idea to run a test with URLs that need no encoding or are already encoded.  They can act as controls to ensure future refactoring efforts do not introduce regressions.

As a slight aside, when adding these test examples, it becomes clear that the "_create_encoded_url" is more of a function that ensures the URL is properly encoded.  Perhaps a name change to something like "_ensure_url_encoding" is appropriate?